### DISCUSSION TITLE
[Q] xlsx read returns "JavaScript heap out of memory"

### DISCUSSION THREAD

**Discussion by sambhav2612** [exceljs/exceljs#2423] [Q&A]
HI,

We are building a data-centric application where CSV and XLSX files are uploaded to the server, saved to a folder by the API controller, read by the API service and converted to a readable stream like: 

```js
const stream = createReadStream(pathname) // where pathname is the absolute path of the file uploaded
```

and this stream is then passed to the workboot instance (which is common for both) like:

```js
      const workbook = new Excel.Workbook()

      let worksheet = null;
      if (file.mimetype === 'text/csv') {
        await workbook.csv.read(stream);
        worksheet = workbook.getWorksheet(1);
      } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet') {
        await workbook.xlsx.read(stream);
        worksheet = workbook.getWorksheet('IMP');
      } else {
        return new Error('Invalid File Format, only .csv/.xlsx allowed!');
      }
```

Now, while I'm able to parse CSVs up to 40MB pretty well, on the same hand while the implementation otherwise is common for both types, XLSXs over 10MB don't parse, the server just gets blocked and after a while returns "JavaScript heap out of memory". 

I just don't understand how this could happen as the module is the same and user API is also pretty much same as well. Maybe under-the-wraps something breaks for larger streams in XLSX parsing. Please help.

**Comment by vunt286** [comment]
@sambhav2612 can you resolve it? I have same issue, pleas help me
